### ISSUE TITLE
Libraries I should not have access to are displayed anyway

### ISSUE THREAD

**Issue by jkuettner** [advplyr/audiobookshelf-app#84]
### Steps to reproduce

1. login as root
1. create a second library
1. create a user who has access only to this new library
1. log in with the new user via the android app
1. the "Main" library appears by default in the top bar, although I don't have access to it

### Expected behaviour

I should only be able to see the libraries that I have access to.

### Actual behaviour

I can see all libraries with any account, even the ones I don't have permission to. The metadata from the libraries I don't have access to (e.g. the cover images) are not loaded, but the titles and the complete library are still displayed:

<img src="https://user-images.githubusercontent.com/12213711/149597747-f42735af-748f-45b7-9042-52e3c2269105.png" width="50%">

The screenshot is from launching the app without navigating.
Actually I should not be allowed to see the `Main` library at all, but the `Main` library seems to be the default setting?

### Environment data

Audiobookshelf Version: `v1.6.57`
App Version: `0.9.37-beta`

- [x] Android App?
- [ ] iOS App?

#### Android Issue
Android version: tested with `12` and `7.0`

Device model: `Google Pixel 4a` and `Honor 5c`

Stock or customized system: `Stock` and `Honor branded`

**Comment by advplyr** [comment]
I was able to resolve this with a server update `v1.6.59`. https://github.com/advplyr/audiobookshelf/commit/8442a629fbc7b16b63cf2c7fdfa11866f03d1354

The api that returns the libraries will now only return the libraries the user has access to.

**Comment by jkuettner** [comment]
Thanks for the fix, it works now via the browser. But I still have the same problem with the app:

If my user only has access to the main library, it now works as it should in the app.

However, if my user only has access to a library that is not main, then the behavior is unfortunately exactly the same as before.

I built and used the Docker image from the current master for testing.

**Comment by advplyr** [comment]
I fixed this now but it requires an update to the mobile app which can take a few days to become available.
In the mobile app the default library was "main", but now it will always set the default library to the first available library. Libraries are orderable.

This also required another server update, I just patched the current version. The server should not have been returning any data when a user requests a library without access. I updated the middleware: https://github.com/advplyr/audiobookshelf/commit/336fbedc759a9c47f20f2d385b844c129bb4c4b5

I'll follow up here when the mobile app is live, thanks for testing.
